src/calculus/AdjointOperator.jl

```diff
@@ -0,0 +1,59 @@
+export AdjointOperator
+
+"""
+`AdjointOperator(A::AbstractOperator)`
+
+Shorthand constructor: 
+
+`'(A::AbstractOperator)`
+
+Returns the adjoint operator of `A`.
+
+```julia
+julia> AdjointOperator(DFT(10))
+ℱᵃ  ℂ^10 -> ℝ^10
+
+julia> [DFT(10); DCT(10)]'
+[ℱ;ℱc]ᵃ  ℂ^10  ℝ^10 -> ℝ^10
+```
+"""
+struct AdjointOperator{T <: AbstractOperator} <: AbstractOperator
+	A::T
+	function AdjointOperator(A::T) where {T<:AbstractOperator} 
+		is_linear(A) == false && error("Cannot transpose a nonlinear operator. You might use `jacobian`")
+		new{T}(A)
+	end
+end
```

Comment:
Do we need this type? There’s already `Adjoint` in Julia, I guess we can use that and the `adjoint` constructor, see https://docs.julialang.org/en/stable/stdlib/LinearAlgebra/#Base.adjoint

Reply:
Not sure we can use Julia's `Adjoint`: its definition reads:
```julia
struct Adjoint{T,S} <: AbstractMatrix{T}
    parent::S
...
```
as you can see its supertype is `AbstractMatrix{T}`. I believe this would restrict us to vectors inputs only.

Concerning the use of `adjoint` as I told you in my last comment we indeed use it already [here](https://github.com/kul-forbes/AbstractOperators.jl/blob/c8314db1840b4d710130829cd3efab94c3ca99a7/src/syntax.jl#L5)